torch/distributions/kl.py

```diff
@@ -6,6 +6,7 @@
 
 import torch
 from torch import inf
+from torch.special import entr
```

Comment:
Should this actually be `torch.special.entr?` or should it be
```
torch.special.xlogy(x, x)
```
What if Tensor is negative? Which is more appropriate?

Reply:
Hm good point, considering a KL divergence $KL(p, q) = p ⋅ \log(p/q)$, which we can rewrite using the [cross entropy](https://en.wikipedia.org/wiki/Cross-entropy) as $KL(p, q) = H(p, q) - H(p)$.

Now, if for some numerical instability reason our algorithm produces $pₖ=-ε$ instead of $pₖ=0$, then:

- with `xlogy` we get `NaN`
- with `entr` we get $+∞$
- true result should be some positive number. (summing over all k)

Seems like the `NaN` would be a better indicator that a problem occurred than $+∞$.